### ISSUE TITLE
Improve documentation for io:sprintf() function

### ISSUE THREAD

**Issue by pubudu91** [ballerina-platform/ballerina-standard-library#559]
The current documentation for `sprintf()` is not comprehensive nor sufficient. Needs considerable improvement on allowed format specifiers and usage.

**Comment by BuddhiWathsala** [comment]
@pubudu91 what kind of information that we need to add here (https://ballerina.io/swan-lake/learn/api-docs/ballerina/io/functions.html#sprintf). I guess most of the important parts are covered.

**Comment by pubudu91** [comment]
This only contains the basics. There seem to be other details related to number formatting etc. For example, this issue was created due to the following question from a user: https://stackoverflow.com/questions/59834650/how-to-do-string-padding-in-ballerina-lang/59836641

Even the sample usage documented in the docs use the following format specifier: `%.2f`. But it doesn't contain any info on what it is or an explanation on how it works. 

Basically the docs should contain comprehensive info on all the features offered by the particular API.
